staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -676,8 +676,12 @@ func (r *Request) Stream() (io.ReadCloser, error) {
 // fn at most once. It will return an error if a problem occurred prior to connecting to the
 // server - the provided function is responsible for handling server errors.
 func (r *Request) request(fn func(*http.Request, *http.Response)) error {
-	//Metrics for total request latency
+	// Metrics for total request latency.
 	start := time.Now()
+
+	// Register metrics for rest client to use.
```

Comment:
This seems more setup than processing. It seems as though this would be better placed in NewRequest() than request(). sync.Once will prevent the additional calls from causing any problems but no point hitting the lock more often than we need to.

Reply:
Good point! I will shift this line to `NewRequest()`. Thanks!